fcm.go

```diff
@@ -209,37 +210,201 @@ func (this *FcmClient) Send() (*FcmResponseStatus, error) {
+	}
+
+	itemIdField, ok := data["item_id"]
+	if ok {
+		itemID = itemIdField.(string)
+	}
+
+	deepLinkField, ok := data["deeplink"]
+	if ok {
+		deepLink = deepLinkField.(string)
+	}
+
+	imageUrlField, ok := data["image_url"]
+	if ok {
+		imageURL = imageUrlField.(string)
+	}
+
+	soundField, ok := data["sound"]
+	if ok {
+		sound = soundField.(string)
+	}
+
+	actorNicknameField, ok := data["actor_nickname"]
+	if ok {
+		actorNickname = actorNicknameField.(string)
+	}
 
+	badgeCountField, ok := data["badge_count"]
+	if ok {
+		badgeCount = int(badgeCountField.(float64))
 	}
-	return this.sendOnce()
 
+	actionsField, ok := data["actions"]
+	if ok {
+		bytes, err := json.Marshal(actionsField)
+		if err == nil {
+			actionsSerialized = string(bytes)
+		}
+	}
+
+	trackingPayloadField, ok := data["tracking_payload"]
+	if ok {
+		bytes, err := json.Marshal(trackingPayloadField)
+		if err == nil {
+			trackingPayloadSerialized = string(bytes)
+		}
+	}
+
+	dataMap := make(map[string]string)
+	dataMap["title"] = title
+	dataMap["body"] = body
+	dataMap["item_type"] = itemType
+	dataMap["item_id"] = itemID
+	dataMap["deeplink"] = deepLink
+	dataMap["image_url"] = imageURL
+	dataMap["sound"] = sound
+	dataMap["actor_nickname"] = actorNickname
+	dataMap["badge_count"] = strconv.Itoa(badgeCount)
+	dataMap["actions"] = actionsSerialized
+	dataMap["tracking_payload"] = trackingPayloadSerialized
+
+	return &dataMap, true
+}
+
+func (this *FcmClient) sendOnceFirebaseAdminGo(client *messaging.Client) (*FcmResponseStatus, error) {
+	multicastMessage, ok := this.Message.makeMulticastMessageData()
+	if !ok {
+		return nil, errors.New("could not build multicast message for Firebase Admin Go library")
+	}
+
+	message := &messaging.MulticastMessage{
+		Data:   *multicastMessage,
+		Tokens: this.Message.RegistrationIds,
+		Notification: &messaging.Notification{
+			Body:     this.Message.Notification.Body,
+			Title:    this.Message.Notification.Title,
+			ImageURL: this.Message.Notification.Icon,
+		},
+	}
+
+	batchResponse, err := client.SendEachForMulticast(context.Background(), message)
+	if err != nil {
+		return &FcmResponseStatus{}, err
+	}
+
+	fcmRespStatus := toFcmRespStatus(batchResponse)
+
+	return fcmRespStatus, nil
 }
 
+func toFcmRespStatus(resp *messaging.BatchResponse) *FcmResponseStatus {
+	var ok bool
+	var statusCode int = http.StatusInternalServerError
 
-// toJsonByte converts FcmMsg to a json byte
-func (this *FcmMsg) toJsonByte() ([]byte, error) {
+	if resp.SuccessCount > 0 {
+		ok = true
+		statusCode = http.StatusOK
+	}
 
-	return json.Marshal(this)
+	status := FcmResponseStatus{
+		Ok:            ok,
+		StatusCode:    statusCode,
+		Success:       resp.SuccessCount,
+		Fail:          resp.FailureCount,
+		Canonical_ids: 0, // TODO Where does it come from? Is it needed?
+		Results:       *toFcmResponseResults(&resp.Responses),
+	}
 
+	return &status
+}
+
+func toFcmResponseResults(original *[]*messaging.SendResponse) *[]map[string]string {
```

Comment:
Same as above. 
Can we please move `toFcmResponseResults` and `toFcmRespStatus` to a different package